packages/web-sdk/src/instrumentations/session/sessionManager.ts

```diff
@@ -0,0 +1,210 @@
+import { dateNow, faro, genShortID, Meta, MetaSession } from '@grafana/faro-core';
+
+import { throttle } from '../../utils';
+import { getItem, isLocalStorageAvailable, removeItem, setItem } from '../../utils/webStorage';
+
+export interface FaroUserSession {
+  sessionId: string;
+  lastActivity: number;
+  started: number;
+  sessionMeta?: MetaSession;
+}
+
+// TODO: make this configurable from the outside
+export const SESSION_EXPIRATION_TIME = 4 * 60 * 60 * 1000; // n hrs
+export const SESSION_INACTIVITY_TIME = 15 * 60 * 1000; // n minutes
+const STORAGE_UPDATE_DELAY = 1 * 1000; // n seconds
+
+export const STORAGE_KEY = '__FARO_SESSION__';
+
+export function createUserSessionObject(sessionId?: string): FaroUserSession {
+  const now = dateNow();
+
+  return {
+    sessionId: sessionId ?? genShortID(),
+    lastActivity: now,
+    started: now,
+  };
+}
+
+export function removeUserSession() {
+  removeItem(STORAGE_KEY);
+}
+
+export function storeUserSession(session: FaroUserSession): void {
+  setItem(STORAGE_KEY, JSON.stringify(session));
+}
+
+export function receiveUserSession(): FaroUserSession | null {
+  const storedSession = getItem(STORAGE_KEY);
+
+  if (storedSession) {
+    return JSON.parse(storedSession) as FaroUserSession;
+  }
+
+  return null;
+}
+
+export function isUserSessionValid(session: FaroUserSession | null): boolean {
+  if (session == null) {
+    return false;
+  }
+
+  const now = dateNow();
+  const lifetimeValid = now - session.started < SESSION_EXPIRATION_TIME;
+  const inactivityPeriodValid = now - session.lastActivity < SESSION_INACTIVITY_TIME;
```

Comment:
nit: You can short circuit this by first checking the lifetimeValid result, quick returning if the session is expired. And then checking the inactivity